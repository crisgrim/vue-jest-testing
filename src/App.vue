<template>
  <div id="app">
    <h1>Testing <strong>Vue</strong> with <em>Jest</em></h1>
    <MessageList :messages="messages"/>
    <Form/>
  </div>
</template>

<script>
import MessageList from './components/MessageList'
import Form from './components/Form'

export default {
  name: 'app',
  data: () => ({ messages: [
    'Key Points of testing',
    'Tools & Configuration',
    'Testing Components',
    'Testing Utils Methods'
  ] }),
  components: {
    MessageList,
    Form
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin: 60px auto;
  overflow: hidden;
  width: 80%;
}
h1 {
    border: 5px solid #2c3e50;
    color: #34495f;
    padding: 20px;
}
h1 strong {
  color: #3ab981;
}

h1 em {
  color: #99424f;
}
</style>
