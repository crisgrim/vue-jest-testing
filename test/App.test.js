import { shallowMount } from '@vue/test-utils'
import App from '../src/App'

describe('App.test.js', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(App);
    cmp.setData({ messages: ['Cat'] });
  });

  it('equals messages to ["Cat"]', () => {
    expect(cmp.vm.messages).toEqual(['Cat']);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});