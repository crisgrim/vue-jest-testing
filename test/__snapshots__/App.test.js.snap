// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <h1>
    Testing 
    <strong>
      Vue
    </strong>
     with 
    <em>
      Jest
    </em>
  </h1>
   
  <messagelist-stub
    messages="Cat"
  />
   
  <form-stub />
</div>
`;
